applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -123,29 +124,49 @@ impl wallet_server::Wallet for WalletGrpcServer {
             .map(|(idx, dest)| -> Result<_, String> {
                 let pk = CommsPublicKey::from_hex(&dest.address)
                     .map_err(|_| format!("Destination address at index {} is malformed", idx))?;
-                Ok((dest.address, pk, dest.amount, dest.fee_per_gram, dest.message))
+                Ok((
+                    dest.address,
+                    pk,
+                    dest.amount,
+                    dest.fee_per_gram,
+                    dest.message,
+                    dest.payment_type,
+                ))
             })
             .collect::<Result<Vec<_>, _>>()
             .map_err(Status::invalid_argument)?;
 
-        let transfers = recipients
-            .into_iter()
-            .map(|(address, pk, amount, fee_per_gram, message)| {
-                let mut transaction_service = self.get_transaction_service();
-                async move {
+        let mut standard_transfers = Vec::new();
+        let mut one_sided_transfers = Vec::new();
+        for (address, pk, amount, fee_per_gram, message, payment_type) in recipients.into_iter() {
+            let mut transaction_service = self.get_transaction_service();
+            if payment_type == PaymentType::StandardMimblewimble as i32 {
+                standard_transfers.push(async move {
                     (
                         address,
                         transaction_service
                             .send_transaction(pk, amount.into(), fee_per_gram.into(), message)
                             .await,
                     )
-                }
-            });
+                });
+            } else if payment_type == PaymentType::OneSided as i32 {
+                one_sided_transfers.push(async move {
+                    (
+                        address,
+                        transaction_service
+                            .send_one_sided_transaction(pk, amount.into(), fee_per_gram.into(), message)
+                            .await,
+                    )
+                });
+            }
```

Comment:
I would probably have just combined the calls in one async block (simpler, no need to allocate any vecs)
```rust
 let transfers = recipients
            .into_iter()
            .map(|(address, pk, amount, fee_per_gram, message, payment_type)| {
                let mut transaction_service = self.get_transaction_service();
                async move {

   if payment_type == PaymentType::StandardMimblewimble as i32 {
                
                    (
                        address,
                        transaction_service
                            .send_transaction(pk, amount.into(), fee_per_gram.into(), message)
                            .await,
                    )
                
            } else if payment_type == PaymentType::OneSided as i32 {
                
                    (
                        address,
                        transaction_service
                            .send_one_sided_transaction(pk, amount.into(), fee_per_gram.into(), message)
                            .await,
                    )
                }
});
```

Reply:
I tried that first but compiler was adamant that the futures were of different types.

Reply:
Are you sure? We arent creating two futures in each of the if-else branches we're creating one future as before that returns the same thing